src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php

```diff
@@ -188,6 +188,7 @@ private function createCloner(): ?ClonerInterface
                 Caster::PREFIX_VIRTUAL.'file' => $flattenException->getFile(),
                 Caster::PREFIX_VIRTUAL.'line' => $flattenException->getLine(),
                 Caster::PREFIX_VIRTUAL.'trace' => new TraceStub($flattenException->getTrace()),
+                Caster::PREFIX_VIRTUAL.'previous' => $flattenException->getPrevious(),
```

Comment:
Will it work this way as `getPrevious()` returns a `FlattenException` instance and not a scalar value? And what about calling `getAllPrevious()` instead?

Reply:
> Will it work this way as getPrevious() returns a FlattenException instance and not a scalar value?

It shows the trace fine in the terminal, and gives all the information needed for further debugging.

Example:
```text
previous: App\Exception\CustomViolationException^ {#1
    message: """
      SQLSTATE[23000]: Integrity constraint violation: ...
      """
    code: 1062
    file: "/app/src/Service/ExampleService.php"
    line: 167
    trace: {
      /app/src/Service/ExampleService.php:167 {
        App\Service\ExampleService->updateStuff(array $updateCollection): void^
        ›
        › throw new CustomViolationException(
        ›     message: $e->getMessage(),
      }
      /app/src/Service/ExampleServiceTwo.php:94 { …}
```

> And what about calling getAllPrevious() instead?

I didn't know about `getAllPrevious()` but after trying it out I don't see much difference between the two except for notation.

`getPrevious`
```text
previous: PDOException^ {
  message: "SQLSTATE[23000]: Integrity constraint violation: ...
}
```

`getAllPrevious`
```text
previousAll: [
  PDOException^ {#3
    message: "SQLSTATE[23000]: Integrity constraint violation: ...
    ...
  },
  PDOException^ {#3}
]
```

As it seems the same to me, what do you think is best to use?